### DISCUSSION TITLE
Navigation voice test

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#106] [Q&A] [answered]
It happens sometimes that the wrong bluetooth channcel is active on the Sena headset or the volume is set to low or not loud enough when driving with ear plugs... then you don't hear the voice commands. 
For the compass tap it could be a new function to play the next navigation instruction. By doing this you have a control that the voice commands is output on the correct channel.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the idea!

We should already have this function:
Tap on the top horizontal navigation panel with the name of the next street and it speaks the next turn instruction.

Does it work for you?

**Reply by DolomitesTom** [reply]
Yes it works, thanks, again learned something 🌝.

**Reply by DolomitesTom** [reply]
When I deactivate street names in settings -->navigation --> display, then this doesn't work any more when tapping on the horizontal panel
(have only next waypoint activated now)

**Reply by devemux86** [reply]
@DolomitesTom 

Indeed, "speak of next turn instruction" currently works with tap on the top "instruction" panel.
We could also attach this function on another more "permanent" panel for more convenience.

Tap on "next waypoint" panel moves temporarily the map on the tapped waypoint.

**Comment by WolfgangZ42** [comment]
Many camera models have two or more buttons that each user can assign selectable functions to. 
Maybe that could be an idea for "our" compass short tap as well?

**Reply by devemux86** [reply]
@WolfgangZ42 very interesting idea!

- We can see how we can offer users to select the functions on the existing buttons / panels
- Or even allow them to add extra buttons / panels with their preferred actions / options?

**Reply by DolomitesTom** [reply]
Sounds good... because this

> Tap on "next waypoint" panel moves temporarily the map on the tapped waypoint.

... is some how not working here.

**Reply by devemux86** [reply]
@DolomitesTom 

> is some how not working here

Do you mean it is not working or is it not very useful?
We can always use a different action for the tap.

**Reply by DolomitesTom** [reply]
> Do you mean it is not working or is it not very useful?

Not working, the map is not moving to this waypoint when tapping.

**Reply by devemux86** [reply]
> the map is not moving to this waypoint when tapping

Are you sure? I cannot reproduce it.
Can you post a route example and the waypoint in the panel?
A screenshot?

**Comment by devemux86** [comment]
> For the compass tap it could be a new function to play the next navigation instruction. By doing this you have a control that the voice commands is output on the correct channel.

Although the compass may not seem the right place to set the voice control,
we can set it there in the next Cruiser update until we can find a better place.

**Reply by WolfgangZ42** [reply]
Sorry, I wouldn't waste a "one-tap position" for the voice control. Maybe this probably rarely needed function could be placed in /tools, like the screen lock ...

**Comment by devemux86** [comment]
> Maybe this probably rarely needed function could be placed in /tools, like the screen lock

@WolfgangZ42 thanks nice idea, we could do that.

(menu is visible when navigation is enabled)

**Comment by devemux86** [comment]
~~The new menu "Tools | Voice test" in navigation is available in [Cruiser 3.0.19](https://play.google.com/store/apps/details?id=gr.talent.cruiser).~~

See https://github.com/devemux86/cruiser/discussions/106#discussioncomment-5964974

**Reply by DolomitesTom** [reply]
don't see it.

**Reply by devemux86** [reply]
Voice test menu is visible when navigation is enabled.

**Reply by DolomitesTom** [reply]
ah ok yes now I see it, but I don't hear anything when tapping on it.

**Comment by DolomitesTom** [comment]
strange after starting new there is no audio message. When I change into simulation and then back into normal, then there is a notification.

**Reply by devemux86** [reply]
Does the navigation direction at the top left show the next turn instruction?
Are you on the route?

Then it can speak the next turn instruction.

**Reply by DolomitesTom** [reply]
no because I had just tried without GPS activation. Then understand how it works and it works.

**Reply by devemux86** [reply]
> I had just tried without GPS activation. 

It currently needs a turn instruction to work.

I can also make it speak something else e.g. "Voice test" and then no need to have active GPS or navigation instructions?

**Reply by DolomitesTom** [reply]
additionally yes, would do so since it is purely a voice test, don't need to be conditioned.

**Comment by devemux86** [comment]
> I can also make it speak something else e.g. "Voice test" and then no need to have active GPS or navigation instructions?

It will be available in the next Cruiser update.

**Comment by devemux86** [comment]
How about moving the "Voice test" in navigation options?
(long press top-left navigation panel)

And we can move the "Screen lock" in navigation options?
(it is also a navigation tool)

So the Tools menu is not so filled with various functions.

**Reply by DolomitesTom** [reply]
Yes, good

**Reply by 0709wiwiwi** [reply]
No new button.
@ start navigation:  "Let's go"  "Let's bike" "Let's drive" etc.
_Nothing new invented but method honestly adopted from another app._ 
 
![image](https://github.com/devemux86/cruiser/assets/54167201/b08b2e8d-c288-43ad-8aa9-41ea7e310c0e)
= "Let's Bike"

**Reply by devemux86** [reply]
The "Voice test" button does not automatically close the screen,
so we can easily press it as many times as we want while testing the sound on our phones and connected devices.

**Reply by 0709wiwiwi** [reply]
>_Nothing new invented but method honestly adopted from another app._

KISS and usefull without extra button.  
As the test button is already there.  
Tap Navigate.  Followed by TTS msg.

> "Let's go" "Let's bike" "Let's drive" etc.

The second useful feature hasn't been noticed? 

> "Cruiser is a map and navigation platform" 

 Auto & Moto & Cycle & Walk. 

Not only does trigger the audio but the message  also confirms that the correct transport means has been selected.
With a properly set up avg speed for the means of transport, it will additionally confirm that a correctly timed announcement and ETA can be expected

**Reply by devemux86** [reply]
Buttons to test (voice) sound are often useful, they already exist in Android and Windows.

Isn't it more convenient to press a button again and again testing any connected devices,
instead of start and stop the navigation (with long press)?

**Reply by 0709wiwiwi** [reply]
A confirmation at the navigation start does not hurt, especially given that extra usefull information spoken.
As with a false router setting the motorbiker can already have to stop and have a correction after the first turn.
Compared with Kurviger motor app Cruiser has to deal with some other disciplines and the best matching set ups.

> start and stop the navigation (with long press)?

Why long press and stop?  
This could just as well be **triggered** after a pause.  No ?  **TAP** >  tap > **TAP** 

_Motorcyclists so do pause the navigation to have a coffee or.......
The bleutooth connection to a helmet may then be broken for a while. 
A disconnected connection will normally resume once back in range.
Surely it is useful after restart of the navigation this is confirmed._

**Reply by WolfgangZ42** [reply]
Using Kurviger/Cruiser since 2018 I never had any problem with voice control ... 🙄

**Reply by 0709wiwiwi** [reply]
> Using Kurviger since 2018 I never had any problem with voice control ... 🙄

Let me guess.....retired since 2018 ?  ;-)

**Reply by WolfgangZ42** [reply]
har har har - I left the office in 2006 ... and don't miss anything! 😛

**Comment by devemux86** [comment, marked as answer]
Voice test is available in navigation options (long press top-left navigation panel) in [Cruiser 3.0.20](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

<img src="https://github.com/devemux86/cruiser/assets/3484020/71f06042-026a-4e2d-af8c-f4efacdc8184" width="50%"/>

**Reply by DolomitesTom** [reply]
Works😁👍

**Reply by devemux86** [reply]
@DolomitesTom thanks for testing it!

Is the spoken text good enough or it could speak something else in the voice test?

**Reply by DolomitesTom** [reply]
What about

"Nice to see you on the road again and using cruiser."

"Schön dass du wieder auf Tour bist und Cruiser benutzt."

😁

**Reply by 0709wiwiwi** [reply]
Very kindly msg, it sounds like a declaration of love, but practically this lacks useful functional value like the currently set mode of transport at the navigation startup.
The message at navigation startup is more optimal with a slightly more businesslike TTS message like "Let's bike" this then for motorbikers. For cyclists, this could then be an expression like "Let's cycle" and so on

**Reply by devemux86** [reply]
@0709wiwiwi 

I can change the message and also announce it on start navigation.

Better be a simpler message, easier to translate into Cruiser's many voice languages.
e.g. "Navigation motorcycle", "Navigation bike", ...

**Reply by WolfgangZ42** [reply]
I'm glad to see, we don't have real problems! 😉

**Reply by 0709wiwiwi** [reply]
And what is your message precisely than saying now Wolfgang ? 
Motorbikers or car drivers that do not change the prefered distance announcements settings are not confronted with false settings when they start up their navigation.  But understand Cruiser is not like the Motorbike Kurviger app but also need to offer optimised distance settings for car or motorbiking and also offer optimised so different distance settings for cyclist and for hiking. If you than start up with a false distance setting by a previously very different activity you than need to halt already at your first very wrongly timed turn announcement.  I confirm you this by experience.  Roger ?

**Reply by devemux86** [reply]
All suggestions that can improve Cruiser are welcome and I note them all!

Don't worry, everything is written on my list and I will definitely check it. :slightly_smiling_face:

**Reply by devemux86** [reply]
In [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser) a short message is spoken when starting the navigation, also containing the selected profile.
e.g. "Navigation, Motorcycle"

**Comment by 0709wiwiwi** [comment]
No problem...most important is that indeed there is a functional message at the navigation launch before the departure.

**Reply by WolfgangZ42** [reply]
Willy,
as this thread is named, "voice test" offers a function to test the voice output of cruiser. Your proposal is to establish a "welcome message" - that's different. Btw: I distinguish between "car" and "motorcycle" via /Settings/Location/Appearance/Location color - and can see easily what I'm using, not only at startup.

**Reply by 0709wiwiwi** [reply]
> Your proposal is to establish a "welcome message" - that's different.  
Different ? You think so ? Read the previous replies, no change in the suggestion
https://github.com/devemux86/cruiser/discussions/106#discussioncomment-5928016
- The "welcome" contains compact info and as a free bonus **also is a usefull voice test.**
> Btw: I distinguish between "car" and "motorcycle"
- Preferences by .pref files are currently a bit awkwardly placed quite deep in the menu. 
- Promoting Presets (done) would make the powerfull .pref files extra usefull and handy.

**Reply by 0709wiwiwi** [reply]
Tested v 3.025 and sure is usefull. Thanks.
